### PR TITLE
Update file opener plugin and add entry points

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: src/fileopener/plugin.ts [hunk 1]
```diff
@@ -92,19 +176,18 @@ class FileOpener implements IFileOpener {
...
     }
 
     // If there we no matches, do nothing.
     if (handlers.length == 0) {
-      return Promise.reject(new Error(`Could not open file '${path}'`));
+      throw new Error(`Could not open file '${path}'`);
 
     // If there was one handler, use it.
     } else if (handlers.length === 1) {
       return this._open(handlers[0], path);
```
Comment (jasongrout) [inline src/fileopener/plugin.ts]: I think if there is no default handler registered, we shouldn't default to the only handler.

**Comment by blink1073** [comment]
Ack, sorry, didn't think any feedback was coming :)

File: src/fileopener/plugin.ts [hunk 2]
```diff
@@ -92,19 +176,18 @@ class FileOpener implements IFileOpener {
...
-    } else if (handlers.length === 0) {
-      for (let h of this._handlers) {
-        if (isDefaultProperty.get(h)) handlers.push(h);
-      }
+    } else if (handlers.length === 0 && this._defaultHandler !== null) {
+       return this._open(this._defaultHandler, path);
     }
 
     // If there we no matches, do nothing.
     if (handlers.length == 0) {
```
Comment (jasongrout) [inline src/fileopener/plugin.ts:184]: This if and the previous if are all testing the same thing (how many handlers we found), and probably should be combined.

**Comment by jasongrout** [comment]
My work depends on yours, so I was reviewing it to (a) be familiar with it and (b) give comments to help it along.  That if statement I pointed about above still is a bit crazy.

**Comment by blink1073** [comment]
I'll make these changes in #16.

**Pull request merged**